src/__tests__/html-text-input-test.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * Internal dependencies
+ */
+import { HTMLInputView } from '../components/html-text-input.js';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+describe( 'HTMLInputView', () => {
+	it( 'HTMLInput renders', () => {
+		const wrapper = shallow(
+			<HTMLInputView />
+		);
+		expect( wrapper ).toBeTruthy();
+	} );
+
+	it( 'HTMLInputView updates store and state on HTML text change', () => {
+		const onChange = jest.fn();
+
+		const wrapper = shallow(
+			<HTMLInputView
+				onChange={ onChange }
+			/>
+		);
+
+		expect( wrapper.instance().state.isDirty ).toBeFalsy();
+
+		// Simulate user typing text
+		const htmlTextInput = findHTMLTextInputFromWrapper( wrapper );
+		htmlTextInput.simulate( 'changeText', 'text' );
+
+		//Check if the onChange is called and the state is updated
+		expect( onChange ).toHaveBeenCalledTimes( 1 );
+		expect( onChange ).toHaveBeenCalledWith( 'text' );
+
+		expect( wrapper.instance().state.isDirty ).toBeTruthy();
+		expect( wrapper.instance().state.value ).toEqual( 'text' );
+	} );
+
+	it( 'HTMLInputView persists changes in HTML text input on blur', () => {
+		const onPersist = jest.fn();
+
+		const wrapper = shallow(
+			<HTMLInputView
+				onPersist={ onPersist }
+				onChange={ jest.fn() }
+			/>
+		);
+
+		// Simulate user typing text
+		const htmlTextInput = findHTMLTextInputFromWrapper( wrapper );
+		htmlTextInput.simulate( 'changeText', 'text' );
+
+		//Simulate blur event
+		htmlTextInput.simulate( 'blur' );
+
+		//Normally prop.value is updated with the help of withSelect
+		//But we don't have it in tests so we just simulate it
+		wrapper.setProps( { value: 'text' } );
+
+		//Check if the onPersist is called and the state is updated
+		expect( onPersist ).toHaveBeenCalledTimes( 1 );
+		expect( onPersist ).toHaveBeenCalledWith( 'text' );
+
+		expect( wrapper.instance().state.isDirty ).toBeFalsy();
+
+		//We expect state.value is getting propagated from prop.value
+		expect( wrapper.instance().state.value ).toEqual( 'text' );
+	} );
+
+	it( 'HTMLInputView propagates title changes to store', () => {
+		const setTitleAction = jest.fn();
+
+		const wrapper = shallow(
+			<HTMLInputView
+				setTitleAction={ setTitleAction }
+			/>
+		);
+
+		// Simulate user typing text
+		const textInput = findTitleTextInputFromWrapper( wrapper );
+		textInput.simulate( 'changeText', 'text' );
+
+		//Check if the setTitleAction is called
+		expect( setTitleAction ).toHaveBeenCalledTimes( 1 );
+		expect( setTitleAction ).toHaveBeenCalledWith( 'text' );
+	} );
+} );
+
+// Utility to find the HTML TextInput from the wrapper
+const findHTMLTextInputFromWrapper = ( wrapper ) => {
```

Comment:
I think we could move those helpers to the top (before the tests), it's a bit confusing having them here imo.
Also the search by attribute is misleading regarding the function name, I would expect a `findByType( TextInput )` maybe?

Reply:
hey @Tug 👋 sure I can move them to the top if that's confusing.

But, I didn't quite follow the misleading part. `Search by attribute` is one way to find the HTML TextInput. There are different queries we can choose to achieve the same thing, but imho this wouldn't affect the function name?

Reply:
Well `findHTMLTextInputFromWrapper` sounds like a generic helper that will find an TextInput element in any kind of wrapper element. So reading the code I thought that what it would do, but jumping at the definition I saw that it was actually matching on a specific attribute of our element. Let's say we want to reuse that function for another wrapper it won't work, or if we change the placeholder value it won't work either, or if for some reason we try to switch language in our tests...
In any case I think it would help if `findHTMLTextInputFromWrapper` was more generic, and maybe we can share those helpers across tests later.

Reply:
I think I got what you mean, you'd expect this function to act as a generic helper but it doesn't currently. Yes actually, I never thought it as a generic helper, this was only intended to be called by different tests from this file only. Could you describe a bit more about the generic helper you were imagining? What kind of query would it have instead?

Reply:
I added a commit to show what I meant: https://github.com/wordpress-mobile/gutenberg-mobile/pull/752/commits/2d45bc0b7f7d99485cda99dfa8aff8e0be923936
I think it's just the naming that was confusing for me, making it clear that the function returns the TextInput content part would have fixed it.

Reply:
thank you @Tug that looks good 👍